pulsar-zookeeper-utils/src/test/java/org/apache/pulsar/zookeeper/ZkIsolatedBookieEnsemblePlacementPolicyTest.java

```diff
@@ -353,4 +353,53 @@ public void testOverlappedBookies() throws Exception {
 
         localZkc.delete(ZkBookieRackAffinityMapping.BOOKIE_INFO_ROOT_PATH, -1);
     }
+    
+    @Test
+    public void testSecondaryIsolationGroupsBookies() throws Exception {
```

Comment:
This test looks great. How about also adding a negative test, that test without `secondaryIsolatedBookieGroup` configured newEnsemble will fail for no enough bookie?